sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -198,6 +209,102 @@
  *    ...
  * }</pre>
  *
+ * <h2>Read from Kafka as a {@link DoFn}</h2>
+ *
+ * {@link ReadSourceDescriptors} is the {@link PTransform} that takes a PCollection of {@link
```

Comment:
We should state the order of preference for how we start reading (offset/timestamp/last commit offset) in this block somewhere.

Reply:
The order is explained in the javadoc of ReadFromKafkaDoFn, [Initial Restriction] section.